nemo/collections/tts/models/fastpitch.py

```diff
@@ -185,24 +219,28 @@ def parser(self):
             )
         return self._parser
 
-    def parse(self, str_input: str) -> torch.tensor:
+    def parse(self, str_input: str, normalize=True, **kwargs) -> torch.tensor:
         if str_input[-1] not in [".", "!", "?"]:
             str_input = str_input + "."
 
+        if normalize:
+            str_input = self.normalizer(str_input)
+
         tokens = self.parser(str_input)
 
         x = torch.tensor(tokens).unsqueeze_(0).long().to(self.device)
         return x
 
+    # T1 - text length, T2 - audio length, T3 - spectrogram length
```

Comment:
~~Where in the code is `T1`, T2`, and `T3` defined?~~ Nvm I see you added it in #3090.
I don't like this naming scheme. Maybe we can do something like T_text? Where T represents AxisKind.Time and everything past _ is ignored.
The other problem with the current implementation is that `T1` and `T2` are both treated as AxisKind.Time so you can pass a dimension of `T1` into a type of `T2`. We ought to differentiate between the two.

Reply:
Fixed it in this PR. I chose chose your idea.